consensus/consensus_fbft.go

```diff
@@ -0,0 +1,30 @@
+package consensus
+
+import "sync"
+
+type LockedFBFTPhase struct {
+	mu     sync.Mutex
+	phrase FBFTPhase
+}
+
+func NewLockedFBFTPhase(initialPhrase FBFTPhase) *LockedFBFTPhase {
+	return &LockedFBFTPhase{
+		phrase: initialPhrase,
+	}
+}
+
+func (a *LockedFBFTPhase) Set(phrase FBFTPhase) {
+	a.mu.Lock()
+	a.phrase = phrase
+	a.mu.Unlock()
+}
+
+func (a *LockedFBFTPhase) Get() FBFTPhase {
+	a.mu.Lock()
+	defer a.mu.Unlock()
+	return a.phrase
+}
```

Comment:
It could be more efficient if we use RWMutex and then using RLock rather than Lock to allow other threads also use Get to read the value .

Reply:
Mostly yes, but not for such simple case. Using rwlock here is over-engineering.

Reply:
If there is no downside or more complexity but only performance benefits for using RWMutex, we should use it.

Reply:
No benefits.
RWMutex is much more complex struct than Mutex, so it takes more time to lock. 
Benefits of shared read will be eaten by getting lock time.